### ISSUE TITLE
hello,I find something wrong in you code..

### ISSUE THREAD

**Issue by weizhao-jiang** [7sDream/torrent_parser#3]
>__version__ = '0.1.2'

When I using you project to parser a torrent file , I got some wrong,here is my found, mybe it can help you ~

* _seek_back nerver rollback _pos
```python3 
def _seek_back(self, count):
        self._content.seek(-count, 1)
        #my_code_here
        self._pos = self._pos - count
```

* the encoding element in torrent files maybe wrong,using charset  instead 
```python3
    def _next_string(self, decode=True):
        length = self._next_int(b':')
        raw = self._read_byte(length)
        if decode:

            #my code here
            self._encoding = chardet.detect(raw)["encoding"]
            if raw == b'':
                string = self._next_string()
            else:
                string = raw.decode(self._encoding)
                
            return string
        return raw
```
* _next_int maybe the negative number.
> like this:  file-mediali-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei-1ei0ei-1ei-1ee

```python3
    def _next_int(self, end=END_INDICATOR):
        
        #以END_INDICATOR为分隔符读出字符串个数
        negative = False
        value = 0
        char = self._read_byte(1)
        while char != end:
            #my_code
            if char == b'-':
                char = self._read_byte(1)
                negative = True 
                
            # noinspection PyTypeChecker
            if not b'0' <= char <= b'9':
                raise InvalidTorrentFileException(self._pos)
            value = value * 10 + int(char) - int(b'0')
            char = self._read_byte(1)
            
        #my_code
        if negative:
            value = -value
        
        return value


```

**Comment by 7sDream** [comment]
Could you please upload some torrent file that lead to those error?  You
can email it to me if you like. Thanks for your help. ：）

ltfychrise <notifications@github.com> 于 2018年4月6日周五 09:43写道：

**Comment by 7sDream** [comment]
I realized that these problems are very easy to reproduce and don't need test files anymore. 

I will fix these issues and add acknowledgments to the ReadMe.

**Comment by 7sDream** [comment]
All three bug/enhancement be fixed/added in version 0.1.4.

**Comment by weizhao-jiang** [comment]
thank you so much  ~
